PR: Automatically unbind conflicting shortcuts when pressing okay in shortcut manager

gepcel opened the pull request:
<!--- Make sure to read the Contributing Guidelines:                   --->
<!--- https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md --->
<!--- and follow PEP 8, PEP 257 and Spyder's code style:               --->
<!--- https://github.com/spyder-ide/spyder/wiki/Dev:-Coding-Style      --->

## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [ ] Included a screenshot (if affecting the UI)

<!--- Explain what you've done and why --->

Adds the ability to automatically unbind a conflicting shortcut when clicking okay in the shortcut manager.

I'm not an expert but want to contribute. So could anyone help me with this?

### Issue(s) Resolved

<!--- List the issue(s) below, in the form "Fixes #1234"; one per line --->

Fixes #8293 


### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct: gepcel

<!--- Thanks for your help making Spyder better for everyone! --->

ccordoba12 commented:
Thanks for your contribution @gepcel!

@goanpeca, please review this one.

gepcel commented:
I've noticed the test failure (before the pull actually). There should be four of them at least about:

```
assert shortcut_editor.warning == SEQUENCE_CONFLICT
assert not shortcut_editor.button_ok.isEnabled()
```

What should I do about them?

pep8speaks commented:
Hello @gepcel! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`spyder/preferences/shortcuts.py`](https://github.com/spyder-ide/spyder/blob/d25e7d67d315f7dc786448145800d5b474aac862/spyder/preferences/shortcuts.py):

> [Line 415:51](https://github.com/spyder-ide/spyder/blob/d25e7d67d315f7dc786448145800d5b474aac862/spyder/preferences/shortcuts.py#L415): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator

CAM-Gerlach commented:
> What should I do about them?

Fix them, to ensure that Ok is enabled and, as necessary, that the other shortcut is unbound. Let us know if you need more specific guidance on this.

gepcel commented:
@CAM-Gerlach ,  tests are fixed. Before it was just me being cautious, feels like need some permission when I changed something and needed to meddle the existed test to let pass.

I actually noticed the dialog height, and googled around a little bit, without a quick answer. I think to enlarge the default height by 3 lines more is not good enough. Since without any conflicts, we should not leave that much empty space, and with more than 1 conflicts, the warning still gets cut out (more than 3 lines). And also, with some shortcut having long context and name, the warning message may be too long to get wrapped into multiple lines. How to the height auto-expand? Could you please help me with this? These pulls recently are my first time dealing with qt.

Actually, while reviewing the code, please consider me as very new and lack of experience on the whole programming and git and pyqt thing, and this my first time digging into spyder source code. I'm not a programmer, only have experience writing scripts with python for data analysis. For example, after pressing a new shortcut, originally the `OK` event is dealted by `def accept(self)`, with a `@Slot` decorator that I don't understand, which is why I added a new func `def unbind_shortcut(self):`. I'm not confident it's appropriate.

ccordoba12 commented:
> How to make the height auto-expand? Could you please help me with this?

@goanpeca, please guide @gepcel to solve this.

gepcel commented:
Now, the height of ShortcutEditor dialog will auto-grow when the warning message gets long. But it won't shrink back when the message gets short after.

![gif](https://user-images.githubusercontent.com/4179106/55387889-d6c78000-5564-11e9-96c9-c83244955951.gif)

ccordoba12 commented on spyder/preferences/shortcuts.py:
```diff
@@ -470,6 +475,13 @@ def unbind_shortcut(self):
         self._qsequences = []
         self.accept()
 
+    def accept_override(self):
```
Please add a docstring to this method.

gepcel replied:
Done.

goanpeca replied:
Please add a period to the end

like `"""Unbind all conflicted shortcuts, and accept the new one."""`

https://www.cockos.com/licecap/

gepcel replied:
I've added a period, but seems too late. Sorry.

goanpeca replied:
Opps xD

goanpeca commented:
@gepcel just one minor comment to fix. Could you make a small screencast of the functionality?

You can use https://www.cockos.com/licecap/

gepcel commented:
> @gepcel just one minor comment to fix. Could you make a small screencast of the functionality?
> 
> You can use https://www.cockos.com/licecap/

Done
